### PR TITLE
[JENKINS-42549, JENKINS-40621] - Handle fallout after #5

### PR DESCRIPTION
**Pull request by oleg-nenashev**
In https://github.com/jenkinsci/lib-jenkins-maven-embedder/pull/5 there was an attempt to fix [JENKINS-40621](https://issues.jenkins-ci.org/browse/JENKINS-40621), but effectively wrong code has been changed. It caused another regression in parallel request handling due to the File session caching in `JarURLConnection` - [JENKINS-42549](https://issues.jenkins-ci.org/browse/JENKINS-42549).

This pull request fixes the regression and the original issue. At least I see no leaked File descriptors after the fix, and the code is working.

@reviewbybees @abayer @aheritier

### CODE REVIEW

File: src/main/java/hudson/maven/MavenEmbedderUtils.java [hunk 1]
```diff
@@ -205,34 +212,39 @@ public static MavenInformation getMavenVersion(File mavenHome) throws MavenEmbed
...
     }
     
+    private static final void closeIt(@CheckForNull Closeable obj) throws MavenEmbedderException {
+        if (obj == null) {
+            return;
+        }
+        try {
+            obj.close();
+        } catch(IOException ex) {
+            throw new MavenEmbedderException("Failed to close " + obj, ex);
```
Comment (jglick) [inline src/main/java/hudson/maven/MavenEmbedderUtils.java]: How is this an improvement over `try`-with-resources?

Comment (oleg-nenashev) [inline src/main/java/hudson/maven/MavenEmbedderUtils.java]: No improvement. The code is just a Java 6 one, hence no try-with-resources

**Comment by oleg-nenashev** [comment]
PR builder is dead. This repo needs a Jenkinsfile hero

**Comment by bitwiseman** [comment]
@oleg-nenashev 
Not much in the way of heroics.  

https://github.com/jenkins-infra/pipeline-library
Example: https://github.com/jenkinsci/git-plugin/blob/master/Jenkinsfile

Then ask (on IRC probably) to have it added to the list of plugins to build

**Comment by oleg-nenashev** [comment]
@bitwiseman Well, I have already added multiple plugins. This guy is a Maven library, which needs some facelifting before putting into CI. Though Jenkinsfile itself is not a big deal

**Comment by bitwiseman** [comment]
@oleg-nenashev - Ah, got it.

File: src/main/java/hudson/maven/MavenEmbedderUtils.java (hunk 1 above)
Comment (jglick) [inline src/main/java/hudson/maven/MavenEmbedderUtils.java]: Maybe update to Java 7?

Comment (oleg-nenashev) [inline src/main/java/hudson/maven/MavenEmbedderUtils.java]: I am fine with that. Should be another PR OTOH, because it will also require changes in dependent plugins

**Comment by aheritier** [comment]
https://ci.jenkins.io/job/Libraries/job/lib-jenkins-maven-embedder/ is ko, I don't know why
https://jenkins.ci.cloudbees.com/job/libraries/job/lib-jenkins-maven-embedder/ I hope this one will be ok

File: src/main/java/hudson/maven/MavenEmbedderUtils.java (hunk 1 above)
Comment (olamy) [inline src/main/java/hudson/maven/MavenEmbedderUtils.java]: java 7! oh yeah baby :-)

**Comment by oleg-nenashev** [comment]
Works on my machine (c)
But seriously, it's really time to write a Pipeline library method for libs based on https://github.com/jenkinsci/pom (which also needs to be facelifted a bit)

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by olamy** [comment]
LGTM

**Comment by oleg-nenashev** [comment]
Thanks! I will spin the release today
